atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/cleaner/DbKvsNamespaceDeleterFactory.java

```diff
@@ -0,0 +1,116 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.dbkvs.cleaner;
+
+import com.google.auto.service.AutoService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.palantir.atlasdb.keyvalue.dbkvs.DbAtlasDbFactory;
+import com.palantir.atlasdb.keyvalue.dbkvs.DbKeyValueServiceConfig;
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleDdlConfig;
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleTableNameGetter;
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleTableNameGetterImpl;
+import com.palantir.atlasdb.keyvalue.dbkvs.cleaner.OracleNamespaceDeleter.OracleNamespaceDeleterParameters;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.ConnectionSupplier;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.OracleDbTableFactory;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.OraclePrefixedTableNames;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.SqlConnectionSupplier;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.TableValueStyleCache;
+import com.palantir.atlasdb.keyvalue.dbkvs.util.DbKeyValueServiceConfigs;
+import com.palantir.atlasdb.keyvalue.dbkvs.util.SqlConnectionSuppliers;
+import com.palantir.atlasdb.namespacedeleter.NamespaceDeleter;
+import com.palantir.atlasdb.namespacedeleter.NamespaceDeleterFactory;
+import com.palantir.atlasdb.spi.KeyValueServiceConfig;
+import com.palantir.atlasdb.spi.KeyValueServiceRuntimeConfig;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import com.palantir.nexus.db.pool.ConnectionManager;
+import com.palantir.nexus.db.pool.HikariClientPoolConnectionManagers;
+import com.palantir.nexus.db.pool.config.ConnectionConfig;
+import com.palantir.nexus.db.pool.config.OracleConnectionConfig;
+import com.palantir.refreshable.Refreshable;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+
+@AutoService(NamespaceDeleterFactory.class)
+public final class DbKvsNamespaceDeleterFactory implements NamespaceDeleterFactory {
+    @Override
+    public String getType() {
+        return DbAtlasDbFactory.TYPE;
+    }
+
+    @Override
+    public NamespaceDeleter createNamespaceDeleter(
+            KeyValueServiceConfig config, Refreshable<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig) {
+        DbKeyValueServiceConfig dbKeyValueServiceConfig = DbKeyValueServiceConfigs.toDbKeyValueServiceConfig(config);
+        String dbType = dbKeyValueServiceConfig.ddl().type();
+
+        if (OracleDdlConfig.TYPE.equals(dbType)) {
+            return createOracleNamespaceDeleter(dbKeyValueServiceConfig, runtimeConfig);
+        }
+        throw new SafeIllegalArgumentException(
+                "Dropping a namespace for the given DB type is not supported",
+                SafeArg.of("type", dbType),
+                SafeArg.of("supportedTypes", OracleDdlConfig.TYPE));
```

Comment:
nit: `List.of(OracleDdlConfig.TYPE)` - seeing a single element be listed as supported types may be confusing